crates/chia-consensus/src/gen/conditions.rs

```diff
@@ -5636,3 +5655,59 @@ fn test_message_eligible_for_ff() {
         assert_eq!((cond.spends[1].flags & ELIGIBLE_FOR_FF), 0);
     }
 }
+
+#[cfg(test)]
+enum DedupCoinOutput {
+    Excess,
+    Equal,
+    Absorb,
+}
+
+#[cfg(test)]
+#[rstest]
+fn test_eligible_for_dedup(
+    #[values(
+        DedupCoinOutput::Excess,
+        DedupCoinOutput::Equal,
+        DedupCoinOutput::Absorb
+    )]
+    output: DedupCoinOutput,
+) {
+    let coin1_output = match output {
+        DedupCoinOutput::Excess => 122,
+        DedupCoinOutput::Equal => 123,
+        DedupCoinOutput::Absorb => 124,
+    };
+
+    let coin2_output = match output {
+        DedupCoinOutput::Excess => 124,
+        DedupCoinOutput::Equal => 123,
+        DedupCoinOutput::Absorb => 122,
+    };
+
+    let test = format!(
+        "(\
+       (({{h1}} ({{h2}} (123 (\
+           ((51 ({{h2}} ({coin1_output} ) \
+           ))\
+       (({{h2}} ({{h1}} (123 (\
+           ((51 ({{h1}} ({coin2_output} ) \
+           ((50 ({{pubkey}} ({{msg1}} ) \
+           ))\
+       ))"
+    );
+
+    let (_a, cond) = cond_test_flag(&test, DONT_VALIDATE_SIGNATURE).expect("cond_test");
+    assert!(cond.spends.len() == 2);
+
+    match output {
+        DedupCoinOutput::Absorb | DedupCoinOutput::Equal => {
+            assert_eq!(cond.spends[0].flags & ELIGIBLE_FOR_DEDUP, 1);
```

Comment:
I'm not excited about making this check depend on the actual flag value of `ELIGIBLE_FOR_DEDUP`